docs/Implementing-easyblocks.rst

```diff
@@ -0,0 +1,486 @@
+
+Note that it is up to you have specific you make the sanity check, but it is recommended to make the check as complete
+as possible to catch any potential build or installation problems thay may occur.
+
+
+.. _implementing_easyblocks_version_specific:
+
+Version-specific parts
+~~~~~~~~~~~~~~~~~~~~~~
+
+In some case, version-specific actions or checks need to be included in an easyblock.
+For this, it is recommended to use ``LooseVersion`` rather than directly comparing version numbers using string values.
+
+For example:
+
+
+.. code:: python
+
+    from distutils.version import LooseVersion
+
+    from easybuild.framework.easyblock import EasyBlock
+
+    class EB_Example(EasyBlock):
+        """Custom easyblock for Example"""
+
+        def sanity_check_step(self):
+            """Custom sanity check for Example."""
+
+            custom_paths = {
+                'files': [],
+                'dirs': [],
+            }
+
+            # in older version, the binary used to be named 'EXAMPLE' rather than 'example'
+            if LooseVersion(self.version) < LooseVersion('1.0'):
+                custom_paths['files'].append('bin/EXAMPLE')
+            else:
+                custom_paths['files'].append('bin/example')
+
+            super(EB_Example, self).sanity_check_step(custom_paths=custom_paths)
+
+
+.. _implementing_easyblocks_module_only_compatibility:
+
+Compatibility with ``--extended-dry-run``/``-x`` and ``--module-only``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Some special care must be taken to ensure that an easyblock is fully compatible with ``--extended-dry-run`` / ``-x``
+(see :ref:`extended_dry_run`) and ``--module-only`` (see :ref:`module_only`).
+
+For ``--extended-dry-run``/``-x``, this is already well covered at :ref:`extended_dry_run_guidelines_easyblocks_detect_dry_run`.
+
+For ``--module-only``, the main thing is to make sure that both the ``make_module_step``, including the ``make_module_*`` submethods,
+and the ``sanity_check_step`` methods do not make any assumptions about the presence of certain environment variables
+or that class variables have been defined already.
+
+This needs to be handled with care since under ``--module-only`` the large majority of the ``*_step`` functions is
+simply skipped entirely. So, if the ``configure_step`` method is responsible for defining class variables that are
+picked up in ``sanity_check_step``, the latter may run into unexpected initial values like ``None``.
+A possible workaround is to define a separate custom method to define the class variables, and to call out to this
+method from ``configure_step`` and ``sanity_check_step`` (for the latter, conditionally, i.e., only if the class
+variables still have the initial values).
+
+
+.. _implementing_easyblocks_using:
+
+Using new/custom easyblocks
+---------------------------
```

Comment:
I feel it's important to highlight this, to avoid that people fiddle with other ways to try and make EasyBuild pick up easyblocks...